File: modules/publix/app/session/batch/akka/actors/services/BatchActionMsgBundle.java
package session.batch.akka.actors.services;

import session.batch.akka.protocol.BatchDispatcherProtocol.BatchActionMsg;

/**
 * Just a basic container for up to three BatchActionMsgs
 * 
 * @author Kristian Lange (2017)
 */
public class BatchActionMsgBundle {

	private BatchActionMsg msg1;
	private BatchActionMsg msg2;
	private BatchActionMsg msg3;

	public static BatchActionMsgBundle build(BatchActionMsg msg1,
			BatchActionMsg msg2, BatchActionMsg msg3) {
		BatchActionMsgBundle bundle = new BatchActionMsgBundle();
		bundle.msg1 = msg1;
		bundle.msg2 = msg2;
		bundle.msg3 = msg3;
		return bundle;
	}

	public static BatchActionMsgBundle build(BatchActionMsg msg1,
			BatchActionMsg msg2) {
		BatchActionMsgBundle bundle = new BatchActionMsgBundle();
		bundle.msg1 = msg1;
		bundle.msg2 = msg2;
		return bundle;
	}

	public static BatchActionMsgBundle build(BatchActionMsg msg1) {
		BatchActionMsgBundle bundle = new BatchActionMsgBundle();
		bundle.msg1 = msg1;
		return bundle;
	}

	public BatchActionMsg[] getAll() {
		if (msg1 == null) {
			return new BatchActionMsg[0];
		} else if (msg2 == null) {
			BatchActionMsg[] a = { msg1 };
			return a;
		} else if (msg3 == null) {
			BatchActionMsg[] a = { msg1, msg2 };
			return a;
		} else {
			BatchActionMsg[] a = { msg1, msg2, msg3 };
			return a;
		}
	}
}

File: modules/publix/app/session/batch/BatchActionMsgBundle.scala
package session.batch

import session.batch.BatchDispatcher.BatchActionMsg

/**
  * Just a basic container for up to three BatchActionMsgs
  *
  * @author Kristian Lange (2017)
  */
class BatchActionMsgBundle(msg1: BatchActionMsg, msg2: BatchActionMsg, msg3: BatchActionMsg) {

  def this(msg1: BatchActionMsg, msg2: BatchActionMsg) = this(msg1, msg2, null)

  def this(msg1: BatchActionMsg) = this(msg1, null, null)

  def getAll: Array[BatchActionMsg] =
    if (msg1 == null) new Array[BatchActionMsg](0)
    else if (msg2 == null) Array(msg1)
    else if (msg3 == null) Array(msg1, msg2)
    else Array(msg1, msg2, msg3)

}
